source/draw/gpu/opengl/OpenGLCanvas.ooc

```diff
@@ -41,10 +41,27 @@ OpenGLCanvas: class extends OpenGLSurface {
 		gpuMap textureTransform = This _createTextureTransform(drawState getSourceNormalized())
 		if (drawState opacity < 1.0f)
 			this context backend blend(drawState opacity)
+		else if (drawState blendMode == BlendMode Add)
+			this context backend blend()
 		else
 			this context backend enableBlend(false)
-		if (drawState inputImage)
-			gpuMap add("texture0", drawState inputImage)
+		tempImageA: GpuImage = null
+		tempImageB: GpuImage = null
+		if (drawState inputImage) {
```

Comment:
The automatic conversion only works for images given as `inputImage` since sending directly to shaders is below the `DrawState` abstraction layer.